src/components/form-draft/publish.vue

```diff
@@ -134,7 +135,11 @@ export default {
           this.versionString !== this.formDraft.version
             ? { version: this.versionString }
             : null
-        )
+        ),
+        fulfillProblem: (problem) => {
+          if (problem.code === 409.6)
+            this.versionConflict = true;
+        }
```

Comment:
The `fulfillProblem` function actually isn't meant to have side effects: rather than being an "on error" callback, it's a way to identify error responses that shouldn't be treated as errors (the function should return a boolean). Such responses will then be handled in the `then()` callback. I think it's best to use promise methods to process the response, but `fulfillProblem()` helps determine which promise callback will be used.

With that in mind, one option is to use `fulfillProblem` + `then()` along these lines:

```js
this.request({
  method: 'POST',
  url: '...',
  fulfillProblem: (problem) => problem.code === 409.6
})
  .then(({ data }) => {
    if (!isProblem(data)) {
      this.$emit('success');
    } else {
      this.$alert().danger(this.$t('problem.409_6'));
      this.versionConflict = true;
    }
  })
  .catch(noop);
```

In this case, `fulfillProblem` identifies a 409.6 Problem as not an error, so it doesn't do the normal error handling, including showing an alert. That's why the `then()` callback here calls `this.$alert()`.

To see another example of `fulfillProblem` + `then()`, check out the `FormNew` component.

I know I suggested that `fulfillProblem` might be the way to approach this issue, but maybe `catch()` would be clearer after all. That way the normal error handling would take place. That would look something like this:

```js
this.request({
  method: 'POST',
  url: '...',
  fulfillProblem: (problem) => problem.code === 409.6
})
  .then(() => {
    this.$emit('success');
  })
  .catch(error => {
    if(error.response != null && isProblem(error.response.data))
      this.versionConflict = true;
  });
```

Also, by the way, if there's a way to improve the comments about `fulfillProblem` in the `request` mixin to make all this clearer, please feel free to do so (either now or later)!

Reply:
I went with the first approach and added an example to the request mixin comment!

Reply:
Awesome, comment is great!